### PR TITLE
Prepare the short code search results screen for mobius migration

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/1955/migrate-short-code-search-results-screen-to-mobius

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt [hunk 1]
```diff
@@ -64,7 +67,15 @@ class ShortCodeSearchResultScreen(context: Context, attributes: AttributeSet) :
...
   private val adapter = ItemAdapter(SearchResultsItemType.DiffCallback())
 
-  private lateinit var disposable: Disposable
+  private val events by unsafeLazy {
+    Observable
+        .merge(
+            searchPatientClicks(),
+            patientItemClicks()
+        )
+        .compose(ReportAnalyticsEvents())
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt:81]: Doesn't the controller already have reporting at this point?

**Review by msasikanth — COMMENTED** [review]
LGTM. Left one comment PTAL

File: app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt:81]: There is no controller in this intermediate architecture. We never ended up reporting these events to analytics at all, looks like.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt:81]: Ah okay 👍🏽

**Review by msasikanth — APPROVED** [review]

**Pull request merged**